x/gov/keeper/deposit.go

```diff
@@ -118,8 +119,54 @@ func (keeper Keeper) AddDeposit(ctx sdk.Context, proposalID uint64, depositorAdd
 		return false, sdkerrors.Wrapf(types.ErrInactiveProposal, "%d", proposalID)
 	}
 
+	// Check coins to be deposited match the proposal's deposit params
+	params := keeper.GetParams(ctx)
+
+	// NOTE: backported from v50
+	// v47 does not have expedited proposals so we always use params.MinDeposit
+	minDepositAmount := params.MinDeposit
+	minDepositRatio, err := sdk.NewDecFromStr(params.GetMinDepositRatio())
+	if err != nil {
+		return false, err
+	}
+
+	if err := keeper.validateDepositDenom(ctx, params, depositAmount); err != nil {
+		return false, err
+	}
```

Comment:
We do have expedited proposals, so we need to include the expedited proposal logic @mattverse

Reply:
Fixed in the most recent commit!